test/integration/etcd/etcd_storage_path_test.go

```diff
@@ -631,6 +656,25 @@ func dumpEtcdKVOnFailure(t *testing.T, kvClient clientv3.KV) {
 	}
 }
 
+func addGVKToEtcdBucket(cohabitatingResources map[string]map[schema.GroupVersionKind]empty, gvk schema.GroupVersionKind, bucket string) {
+	if cohabitatingResources[bucket] == nil {
+		cohabitatingResources[bucket] = map[schema.GroupVersionKind]empty{}
+	}
+	cohabitatingResources[bucket][gvk] = empty{}
+}
+
+func getEtcdBucket(path string) string {
```

Comment:
comment on what this is doing (stripping last segment, assuming it's an object name segment) would be helpful